src/Routing/Route/Route.php

```diff
@@ -187,9 +187,9 @@ protected function normalizeAndValidateMethods($methods)
     {
         $methods = is_array($methods)
             ? array_map('strtoupper', $methods)
-            : strtoupper($methods);
+            : [strtoupper($methods)];
 
-        $diff = array_diff((array)$methods, static::VALID_METHODS);
+        $diff = array_diff($methods, static::VALID_METHODS);
```

Comment:
Just cast it to an array in array_map.

Reply:
I was trying to avoid always casting. The current code will only create an array if needed, otherwise there is no cast. From what I remember array_map is not as fast as creating a new array with the strtoupper result.